jujugui/static/gui/src/app/test-init.js

```diff
@@ -0,0 +1,101 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+'use strict';
+
+const Keysim = require('keysim');
+const utils = require('../test/utils');
+
+describe('init', () => {
+  let app, container, JujuGUI;
+
+  const createApp = config => {
+    const defaults = {
+      apiAddress: 'http://api.example.com/',
+      bundleServiceURL: 'http://examle.com/bundleservice',
+      controllerSocketTemplate: 'wss://$server:$port/api',
+      socket_protocol: 'wss',
+      baseUrl: 'http://example.com/',
+      charmstoreURL: 'http://1.2.3.4/',
+      flags: {},
+      gisf: false,
+      plansURL: 'http://plans.example.com/',
+      termsURL: 'http://terms.example.com/'
+    };
+    // Overwrite any default values with those provided.
+    const initConfig = Object.assign(defaults, config);
+    return new JujuGUI(initConfig);
+  };
+
+  beforeAll(done => {
+    // The module list is copied from index.html.mako and is required by
+    // init.js.
+    YUI(GlobalConfig).use([
```

Comment:
Maybe this array should be moved to a separate file and included in the index's so we don't have to try and keep them all in sync?